debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbErrorHandler.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import io.debezium.connector.base.ChangeEventQueue;
+import io.debezium.pipeline.ErrorHandler;
+
+/**
+ * Error handler for MongoDB.
+ *
+ * @author John Graf
+ */
+public class MongoDbErrorHandler extends ErrorHandler {
+
+    public MongoDbErrorHandler(String logicalName, ChangeEventQueue<?> queue) {
+        super(MongoDbConnector.class, logicalName, queue);
+    }
+
+    @Override
+    protected boolean isRetriable(Throwable throwable) {
+        if (throwable instanceof org.apache.kafka.connect.errors.ConnectException) {
```

Comment:
That's a bit too coarse-grained, we'd e.g. also retry in case of wrong credentials or other cases where retries are unlikely to help. Also we want to move away from raising `ConnectException` internally; could you narrow this down to specific MongoDB exception types and/or messages which make it reasonable to do a retry?

Reply:
The problem is that {{ConnectException}} is wrapping the MongoDB exceptions. Are you suggesting that we look at {{throwable.getCause()}}?

Reply:
That would be an option, yes.

Reply:
OK, I submitted an update that loops through getCause() looking for a MongoSocketException.